node/src/database/db_migrations.rs

```diff
@@ -169,124 +184,96 @@ impl DbMigratorReal {
         mut migration_utilities: Box<dyn DBMigrationUtilities + 'a>,
         list_of_updates: &'a [&'a (dyn DatabaseMigration + 'a)],
     ) -> Result<(), String> {
-        let updates_to_process =
-            Self::select_updates_to_process(mismatched_schema, list_of_updates);
-        let mut peekable_list = updates_to_process.iter().peekable();
-        for _ in 0..peekable_list.len() {
-            let (next_record, updatable_to) = Self::process_items_with_dirty_references(
-                peekable_list.next(),
-                peekable_list.peek(),
-            );
-            let current_state = next_record.old_version();
+        let updates_to_process = Self::select_updates_to_process(
+            mismatched_schema,
+            list_of_updates,
+            &*migration_utilities,
+        );
+        for record in updates_to_process {
+            let current_version = record.old_version();
             //necessary for testing
-            if Self::is_target_version_reached(current_state, &target_version) {
+            if Self::is_target_version_reached(&current_version, &target_version) {
```

Comment:
This may no longer be required at all, since the for loop terminates after the last migration in the list has been processed. If you need to execute just some of these updates for testing, perhaps you could extract the guts of this for loop into another function, and then call that function a certain number of times from the test.